### PR TITLE
SVCD::unsubscribe [NOT READY]

### PR DESCRIPTION
**Pull request by nalnaji**

### CODE REVIEW

**Comment by mccormickt12** [comment]
wow this looks great!

-TBD

File: natlib/svcd.c [hunk 1]
```diff
@@ -185,3 +186,41 @@ static int svcd_init( lua_State *L )
...
+
+    //send msg over socket
+    lua_pushlightfunction(L, libstorm_net_sendto);
+    lua_pushstring(L, "ncsock");
+    lua_gettable(L, 5);
+    //msg as str
+    lua_pushlightfunction(L, arr_as_str);
+    uint8_t msg[7];
+    msg[0] = 0;
+    ((uint16_t*) (msg+1))[0] = svcid;
```
Comment (immesys) [inline natlib/svcd.c:217]: Something is still fishy here. Firstly you are overwriting locations, and secondly you ignored my comment about unaligned writes. This is unsafe code.

File: natlib/svcd.c [hunk 2]
```diff
@@ -185,3 +186,41 @@ static int svcd_init( lua_State *L )
...
+    ((uint16_t*) (msg+1))[1] = attrid;
+    ((uint16_t*) (msg+1))[0] = luaL_checknumber(L, 2); //svcid
```
Comment (immesys) [inline natlib/svcd.c:219]: checknumber is _ONLY_ for arguments

**Comment by immesys** [comment]
Not ready yet. You need to fix your unaligned writes.

**Comment by immesys** [comment]
superseded by #44

**Pull request closed without merging**